network/join_test.go

```diff
@@ -39,96 +39,35 @@ func TestJoin(t *testing.T) {
-		suite.CosmosClientMock.
-			On(
-				"BroadcastTx",
-				context.Background(),
-				account,
-				launchtypes.NewMsgSendRequest(
-					addr,
-					testutil.LaunchID,
-					launchtypes.NewGenesisValidator(
-						testutil.LaunchID,
-						addr,
-						gentx.JSON(t),
-						[]byte{},
-						sdk.NewCoin(TestDenom, sdkmath.NewInt(TestAmountInt)),
-						launchtypes.Peer{
-							Id: testutil.NodeID,
-							Connection: &launchtypes.Peer_TcpAddress{
-								TcpAddress: testutil.TCPAddress,
-							},
-						}),
-				),
-			).
-			Return(testutil.NewResponse(&launchtypes.MsgSendRequestResponse{
-				RequestID:    TestGenesisValidatorRequestID,
-				AutoApproved: false,
-			}), nil).
-			Once()
+		expectedReqs := []launchtypes.RequestContent{
+			launchtypes.NewGenesisValidator(
+				testutil.LaunchID,
+				addr,
+				gentx.JSON(t),
+				[]byte{},
+				sdk.NewCoin(TestDenom, sdkmath.NewInt(TestAmountInt)),
+				launchtypes.Peer{
+					Id: testutil.NodeID,
+					Connection: &launchtypes.Peer_TcpAddress{
+						TcpAddress: testutil.TCPAddress,
+					},
+				},
+			),
+		}
 
-		joinErr := network.Join(
+		suite.ChainMock.On("NodeID", context.Background()).Return(testutil.NodeID, nil).Once()
+		reqs, err := network.GetJoinRequestContents(
 			context.Background(),
 			suite.ChainMock,
 			testutil.LaunchID,
 			gentxPath,
 			WithPublicAddress(testutil.TCPAddress),
 		)
-		require.NoError(t, joinErr)
-		suite.AssertAllMocks(t)
-	})
-
-	t.Run("successfully send join request with custom gentx", func(t *testing.T) {
-		account := testutil.NewTestAccount(t, testutil.TestAccountName)
-		tmp := t.TempDir()
-		addr, err := account.Address(networktypes.SPN)
 		require.NoError(t, err)
-		gentx := testutil.NewGentx(
-			addr,
-			TestDenom,
-			TestAmountString,
-			"",
-			testutil.PeerAddress,
-		)
-		gentxPath := gentx.SaveTo(t, tmp)
-		suite, network := newSuite(account)
-
-		suite.CosmosClientMock.
-			On(
-				"BroadcastTx",
-				context.Background(),
-				account,
-				launchtypes.NewMsgSendRequest(
-					addr,
-					testutil.LaunchID,
-					launchtypes.NewGenesisValidator(
-						testutil.LaunchID,
-						addr,
-						gentx.JSON(t),
-						[]byte{},
-						sdk.NewCoin(TestDenom, sdkmath.NewInt(TestAmountInt)),
-						launchtypes.Peer{
-							Id: testutil.NodeID,
-							Connection: &launchtypes.Peer_TcpAddress{
-								TcpAddress: testutil.TCPAddress,
-							},
-						},
-					),
-				),
-			).
-			Return(testutil.NewResponse(&launchtypes.MsgSendRequestResponse{
-				RequestID:    TestGenesisValidatorRequestID,
-				AutoApproved: false,
-			}), nil).
-			Once()
-
-		joinErr := network.Join(context.Background(), suite.ChainMock, testutil.LaunchID, gentxPath)
-		require.NoError(t, joinErr)
-		suite.AssertAllMocks(t)
+		require.ElementsMatch(t, expectedReqs, reqs)
 	})
 
-	t.Run("failed to send join request, failed to broadcast join tx", func(t *testing.T) {
+	t.Run("successfully get join request with custom gentx", func(t *testing.T) {
```

Comment:
I don't spot any difference between this case and the one before. Both seems to use the same kind of `gentx`. Hence I don't understand why this case is labelled *with custom gentx*.

Reply:
I think there is some tricky code modification with the delta of the pull request.
I removed "failed to send join request, failed to broadcast join tx" (because the method no longer broadcast tx anymore) and kept "successfully send join request with custom gentx" but it seems the PR detect like I removed the second and modified the body of the first actually removed

Reply:
Actually I compared the 2 test cases with the source file directly, not the diff (which isn't really readable).
![image](https://user-images.githubusercontent.com/92280/211028640-92e49dea-fb43-491a-93cb-2059f50c890f.png)

Reply:
Seems like the difference is not on the gentx, but on `WithPublicAddress` which is only present in the first test case.

Reply:
Yes, you are right.

I think what happened is we changed the behavior of the method in the past without changing the test case name.
If I remember well, we used to have an option to specify the path of a custom gentx (otherwise a default path was used) then we made the gentx path arg mandatory but give the option to provide a custom peer address

Reply:
I changed test names